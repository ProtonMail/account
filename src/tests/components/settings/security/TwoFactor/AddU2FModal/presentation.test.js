import render from 'preact-render-to-string';
import { deep } from 'preact-render-spy';

import Presentation from '../../../../../../components/settings/security/TwoFactor/AddU2FModal/presentation';

describe('AddU2FModal Presentation step', () => {
    test('regular display', () => {
        expect(render(<Presentation/>)).toMatchSnapshot();
    });
    test('display with message', () => {
        expect(render(<Presentation message='This is a message'/>)).toMatchSnapshot();
    });

    test('submit and cancel', () => {
        const onSubmit = jest.fn();
        const onCancel = jest.fn();
        const context = deep(<Presentation onSubmit={onSubmit} onCancel={onCancel}/>);

        const event = { preventDefault: () => undefined };

        context.find('form').simulate('submit', event);
        expect(onSubmit).toHaveBeenCalled();
        expect(onCancel).not.toHaveBeenCalled();

        context.find('form').simulate('reset', event);
        expect(onSubmit).toHaveBeenCalledTimes(1); // the submit event
        expect(onCancel).toHaveBeenCalled(); // only once
    });
});
