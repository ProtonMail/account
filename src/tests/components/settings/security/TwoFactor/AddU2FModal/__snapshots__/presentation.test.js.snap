// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddU2FModal Presentation step display with message 1`] = `
<form class="wrapper">
  <div class="content ">
    <div>This is a message</div>
  </div>
  <div class="footer">
    <button type="reset" value="Reset">Back</button>
    <button type="submit" value="Submit">Next</button>
  </div>
</form>
`;

exports[`AddU2FModal Presentation step regular display 1`] = `
<form class="wrapper">
  <div class="content ">
    <p>This wizard will add a new security key to your Proton Account.</p>
    <p>Please note that you will not be able to access your account if you loose
      your U2F device and your recovery codes. We recommend setting up a second
      2FA method as a backup.</p>
    <p>
      <strong>If you have never used 2FA before, we strongly recommend you reading our
        2FA Guide first.</strong>
    </p>
    <div>
      <a class="button" href="https://protonmail.com/blog/" target="_blank"
      rel="noopener noreferrer">READ 2FA GUIDE</a>
    </div>
  </div>
  <div class="footer">
    <button type="reset" value="Reset">Back</button>
    <button type="submit" value="Submit">Next</button>
  </div>
</form>
`;
